src/UiSelect.vue

```diff
@@ -237,10 +241,11 @@ export default {
             query: '',
             isActive: false,
             isTouched: false,
-            selectedIndex: -1,
-            highlightedIndex: -1,
```

Comment:
Instead of tracking indexes, I now track the actual option. The caveat is that if there is more than one option with the same value, they will both be marked as selected.